src/test/scala/com/cognite/sdk/scala/v1/FilesTest.scala

```diff
@@ -243,4 +256,30 @@ class FilesTest extends SdkTestSpec with ReadBehaviours with WritableBehaviors w
     assert(out.toByteArray().sameElements(expected))
     client.files.deleteById(file.id)
   }
+
+  it should "support search with dataSetIds" in {
+    val created = filesToCreate.map(f => client.files.createOneFromRead(f))
+    try {
+      val fromTime = created.map(_.createdTime).min
+      val toTime = created.map(_.createdTime).max
+      val foundItems = retryWhileEmpty {
+        client.files.search(FilesQuery(Some(FilesFilter(
+          dataSetIds = Some(Seq(CogniteInternalId(testDataSet.id))),
+          createdTime = Some(TimeRange(
+            min=fromTime,
+            max=toTime
+          ))
+        ))))
+      }
+      assert(!foundItems.isEmpty)
+      foundItems.foreach({ i =>
```

Comment:
`assert(foundItems.forall(_.dataSetId.contains(testDataSet.id)))` but even better is `foundItems.map(_.dataSetId) should contain only (Some(testDataSet.id))`, not only because it more clearly expresses the intent but also because it will give much more helpful error messages.

See http://www.scalatest.org/user_guide/using_matchers#workingWithContainers
and http://www.scalatest.org/user_guide/using_matchers#workingWithAggregations